gguf-py/gguf/gguf.py

```diff
@@ -545,15 +576,16 @@ def add_tensor_info(self, name: str, tensor_shape: np.ndarray, tensor_dtype: np.
         self.ti_data_count += 1
 
     def add_tensor(self, name: str, tensor: np.ndarray, raw_shape: Optional[np.ndarray] = None, raw_dtype: Optional[GGMLQuantizationType] = None):
-        if self.use_temp_file and not hasattr(self, "temp_file"):
-            self.temp_file = tempfile.SpooledTemporaryFile(mode="w+b", max_size=256*1024*1024)
-            self.temp_file.seek(0)
+        if self.use_temp_file and self.temp_file is None:
+            fp = tempfile.SpooledTemporaryFile(mode="w+b", max_size=256*1024*1024)
+            fp.seek(0)
+            self.temp_file = fp
 
         self.add_tensor_info(name, raw_shape if raw_shape is not None else tensor.shape, tensor.dtype, tensor.nbytes, raw_dtype = raw_dtype)
```

Comment:
There's a type error here still. `add_tensor_info` wants the shape to be an `np.ndarray` but typechecking thinks `somearray.shape` is type `object`. Not completely sure how to fix it.

Reply:
The type checker (I'm using mypy) probably doesn't handle the ternary the way you would like - it's just coalescing the resulting type to 'object' because there is no overlap. Change it to a full 'if' statement with a union for the result type and it will be more accurate - although the signature of 'add_tensor_info' will have to be changed to accept a tuple for 'tensor_shape'.

Reply:
Thanks. I'll try to that tomorrow. I think there's a generic `Sequence` type that works for tuples, lists, etc. Not sure if it works for ndarrays too.